packages/react/src/components/UIShell/__tests__/HeaderPanel-test.js

```diff
@@ -63,5 +64,38 @@ describe('HeaderPanel', () => {
       const childrenArray = screen.getAllByText('Test');
       expect(childrenArray.length).toEqual(2);
     });
+
+    it('should call `onHeaderPanelFocus` callback, when defined', async () => {
+      const callback = jest.fn();
+      render(
+        <HeaderPanel onHeaderPanelFocus={callback}>
+          <button data-testid="button" type="button">
+            Test
+          </button>
+        </HeaderPanel>
+      );
+
+      screen.getByTestId('button').focus();
+      await userEvent.keyboard('{Escape}');
+
+      await waitFor(() => expect(callback).toHaveBeenCalled());
+    });
+
+    it('should not error when `onHeaderPanelFocus` is not defined', async () => {
+      render(
+        <HeaderPanel>
+          <button data-testid="button" type="button">
+            Test
+          </button>
+        </HeaderPanel>
+      );
+
+      await waitFor(() =>
+        expect(async () => {
+          screen.getByTestId('button').focus();
+          await userEvent.keyboard('{Escape}');
+        }).not.toThrow()
+      );
+    });
```

Comment:
Some minor nitpicks, suggested via the change below

- rename `callback` to `onHeaderPanelFocus`
- refactor to remove waitFor (these shouldn't be necessary I don't think)
- use accessible queries instead of `getByTestId`

```suggestion
    it('should call `onHeaderPanelFocus` callback, when defined', async () => {
      const onHeaderPanelFocus = jest.fn();
      render(
        <HeaderPanel onHeaderPanelFocus={onHeaderPanelFocus}>
          <button type="button">Test</button>
        </HeaderPanel>
      );

      screen.getByRole('button', { name: 'Test' }).focus();
      await userEvent.keyboard('{Escape}');

      expect(onHeaderPanelFocus).toHaveBeenCalled();
    });

    it('should not error when `onHeaderPanelFocus` is not defined', async () => {
      render(
        <HeaderPanel>
          <button type="button">Test</button>
        </HeaderPanel>
      );

      await expect(async () => {
        screen.getByRole('button', { name: 'Test' }).focus();
        await userEvent.keyboard('{Escape}');
      }).not.toThrow();
    });
```

Reply:
@sjbeatle I went ahead and pushed up this suggested change to this PR, hope that's okay, let me know if there's something I missed!